### DISCUSSION TITLE
npm audit not catching mpath vulnerability

### DISCUSSION THREAD

**Discussion by zademnianiuk** [npm/feedback#562] [CLI] [answered]
## Problem Statement
I'm using [trivy](https://github.com/aquasecurity/trivy) to scan my image for vulnerabilities. It's telling me I have the following vulnerability in my `package-lock.json`:
```
<my_repo>:<my_tag> (alpine 3.14.2)
=============================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)


home/node/package-lock.json (npm)
=================================
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 1)

+---------+------------------+----------+-------------------+---------------+---------------------------------------+
| LIBRARY | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                 TITLE                 |
+---------+------------------+----------+-------------------+---------------+---------------------------------------+
| mpath   | CVE-2021-23438   | CRITICAL | 0.8.1             | 0.8.4         | mpath: type confusion can lead        |
|         |                  |          |                   |               | to a bypass of CVE-2018-16490         |
|         |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2021-23438 |
+---------+------------------+----------+-------------------+---------------+---------------------------------------+
```

But when running `npm audit` it doesn't seem to find it:
```
{
  "auditReportVersion": 2,
  "vulnerabilities": {},
  "metadata": {
    "vulnerabilities": {
      "info": 0,
      "low": 0,
      "moderate": 0,
      "high": 0,
      "critical": 0,
      "total": 0
    },
    "dependencies": {
      "prod": 205,
      "dev": 964,
      "optional": 36,
      "peer": 0,
      "peerOptional": 0,
      "total": 1180
    }
  }
}
```

Looking at that `mpath` vulnerability in the National Vulnerability Database ([reference](https://nvd.nist.gov/vuln/detail/CVE-2021-23438#range-6911880)), it looks like it was published `09/01/2021`.

Looking at the [NPM Advisories](https://www.npmjs.com/advisories), it looks like the most recent one was from `08/31/2021`.

## Potential Solution
Does this vulnerability just need to be added to the NPM Advisory database?

Thanks in advance!

**Comment by MylesBorins** [comment, marked as answer]
Thanks for catching this... we have manually update the advisory DB

https://www.npmjs.com/advisories/1785

We are working on something to help automate this process that will keep things in sync and hoping to roll it out in the near future
